ARCharts/ARBarChart.swift

```diff
@@ -33,57 +33,107 @@ public class ARBarChart: SCNNode {
      */
     public init(dataSource: ARBarChartDataSource, delegate: ARBarChartDelegate, size: SCNVector3) {
         super.init()
+        
         self.dataSource = dataSource
         self.delegate = delegate
-        
-        // Compute normalization constants to scale the bars to fit inside the bounding box
-        let xRange = dataSource.numberOfSeries(in: self)
-        let yRange = Array(0 ..< xRange).map({ dataSource.barChart(self, numberOfValuesInSeries: $0) }).max() ?? 1
-        // TODO: Should we always let zMin be 0? I.e. minimum bar should always have some depth, right?
-        var zMin = 0.0
-        var zMax = 0.0
-        for xValue in 0 ..< dataSource.numberOfSeries(in: self) {
-            for yValue in 0 ..< dataSource.barChart(self, numberOfValuesInSeries: xValue) {
-                let zValue = dataSource.barChart(self, valueAtIndex: yValue, forSeries: xValue)
-                if zValue < zMin {
-                    zMin = zValue
-                }
-                if zValue > zMax {
-                    zMax = zValue
+        drawGraph(withSize: size)
+    }
+    
+    public func reloadGraph() {
+        // TODO: Implement
+    }
+    
+    // TODO: Cache (with lazy?)
+    private var numberOfSeries: Int? {
+        get {
+            return self.dataSource?.numberOfSeries(in: self)
+        }
+    }
+    
+    // TODO: Cache (with lazy?)
+    private var maxNumberOfIndeces: Int? {
+        get {
+            guard let numberOfSeries = self.numberOfSeries, let dataSource = self.dataSource else {
+                return nil
+            }
+            
+            return Array(0 ..< numberOfSeries).map({ dataSource.barChart(self, numberOfValuesInSeries: $0) }).max()
+        }
+    }
+    
+    // TODO: Cache (lazy?)
```

Comment:
Yes, I think this one should be a `lazy var`, since we might have a huge number of values to loop over. Also might call this multiple times if re-rendering and changing the bounding box.